entrypoint/microservice/routes/documents.py

```diff
@@ -0,0 +1,219 @@
+        )
+
+    success, output = get_documents_from_db(document_ids.split(','))
+    if success:
+        return jsonify(output), 200
+    else:
+        return jsonify(output), 400
+
+@bp.route('/<doc_id>', methods=['GET'])
+def retrieve_document(doc_id):
+    """
+    At this endpoint you will retrieve a single document, if it exist. GET method on route 
+    documents/<document_id> will return you the JSON of the document of the specified id.
+    """
+
+    success, output = get_documents_from_db([doc_id])
+    if success:
+        return jsonify(output), 200
+    else:
+        return jsonify(output), 400
+
+@bp.route('/<doc_id>/similar', methods=['GET'])
+def get_similar_documents(doc_id):
+    """
+    At this endpoint you will be able to get documents similar to document `doc_id`.
+    You can provide additional query_parameter:
+        * get_k int (number of results), default = 5
+    
+    In response you will receive json of the following format:
+
+    {
+    "finish": true,
+    "similar_documents": [
+        1000017599,
+        1000017600,
+        1000017598,
+        1000017597,
+        1000017596,
+        1000017593,
+        1000017595,
+        1000017594
+    ],
+    "similarities": [
+        [
+            1000017599,
+            0.199445346504904
+        ],
+        [
+            1000017600,
+            0.19275458304224
+        ],
+        [
+            1000017598,
+            0.191803893650522
+        ],
+        [
+            1000017597,
+            0.190294593704065
+        ],
+        [
+            1000017596,
+            0.190035190653879
+        ],
+        [
+            1000017593,
+            0.189315287176367
+        ],
+        [
+            1000017595,
+            0.189190944259772
+        ],
+        [
+            1000017594,
+            0.184603884562228
+        ]
+    ]
+    }
+
+    Example request
+    {BASE_URL}/api/v1/documents/123?get_k=5
+    will return top 5 similar documents to document with id 123.
+    """
+
+    HOST = app.config.get('SIMILARITY_HOST')
+    PORT = app.config.get('SIMILARITY_PORT')
+    
+    query_params = {
+        'document_id' : doc_id,
+        'get_k' : request.args.get('get_k', 5)
+    }
+    
+    r = requests.get(f"http://{HOST}:{PORT}/api/v1/similarity/get_similarities", params=query_params)
+    return jsonify(r.json())
+
+@bp.route('/<doc_id>/similarity_update', methods=['POST'])
+def update_document_similarities(doc_id):
+    """
+    Make a request with GET method to this endpoint.
+
+    Example request:
+    {BASE_URL}/api/v1/documents/id/similarity_update
+    """
+
+    HOST = app.config.get('SIMILARITY_HOST')
+    PORT = app.config.get('SIMILARITY_PORT')
+    query_params = {
+        'document_id': doc_id,
+    }
+    r = requests.get(f"http://{HOST}:{PORT}/api/v1/similarity/new_document_embedding", params=query_params)
+    return jsonify(r.json())
+
+@bp.route('/find', methods=['GET'])
```

Comment:
Maybe `/search` would be a better naming? I think it is more commonly used than `/find`.